lib/levanter/src/levanter/models/olmo.py

```diff
@@ -609,3 +611,416 @@ def resize_vocab(self, new_size: int, key=None) -> "LmHeadModel[Olmo2Config]":
             return dataclasses.replace(self, embeddings=new_embeddings, lm_head=new_lm_head)
         else:
             return dataclasses.replace(self, embeddings=new_embeddings)
+
+
+silence_transformer_nag()
+from transformers import PretrainedConfig as HfConfig  # noqa: E402
+
+
+# ------------------------------------------------------------
+# Sliding Window Mask
+# ------------------------------------------------------------
+
+def build_sliding_causal_mask(Pos, KeyPos, window):
```

Comment:
the AttentionMask class already has sliding windows implemented. We should be using that: https://github.com/marin-community/marin/blob/14c1488198804b998b24fc6ba58340c2ce20ea04/lib/levanter/src/levanter/layers/attention.py#L956-L964